Change ble_driver queue.Queue to queue.SimpleQueue

grandcentrixjosh opened the pull request:
Queue.put() is not reentrant. That leads to race
conditions, especially on resource constrained
systems. In this lib this leads to Gatt Attributes not being discovered properly.
SimpleQueue.put() is reentrant and solves those problems.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/NordicSemiconductor/pc-ble-driver-py?pullRequest=232) <br/>All committers have signed the CLA.

grandcentrixjosh committed:
Change ble_driver queue.Queue to queue.SimpleQueue

Queue.put() is not reentrant. That leads to race
conditions, especially on resource constrained
systems. In this lib this leads to Gatt Attributes not being discovered
properly.
SimpleQueue.put() is reentrant and solves those problems.
